go/test/endtoend/recovery/unshardedrecovery/recovery.go

```diff
@@ -91,16 +92,19 @@ func TestMainImpl(m *testing.M) {
 		}
 		localCluster.Keyspaces = append(localCluster.Keyspaces, *keyspace)
 
+		// Create a new init_db.sql file that sets up passwords for all users.
+		// Then we use a db-credentials-file with the passwords.
+		// We need these users during backup-restore process for example vt_repl is needed during ResetReplication while taking backup.
 		dbCredentialFile = cluster.WriteDbCredentialToTmp(localCluster.TmpDirectory)
 		initDb, _ := os.ReadFile(path.Join(os.Getenv("VTROOT"), "/config/init_db.sql"))
 		sql := string(initDb)
+		// since original initDb does not have updated passwords. Therefore, update the init file with passwords SQL.
+		oldAlterTableMode := `SET GLOBAL old_alter_table = ON;`
```

Comment:
I wish we had a code comment as to why we want `old_alter_table` to be ON. I'm sure there was a good reason.

Reply:
yeah , I didn't research that part..

Reply:
https://github.com/vitessio/vitess/issues/8315, its mainly due to xtrabackup